google/resource_compute_region_instance_group_manager.go

```diff
@@ -145,6 +149,18 @@ func resourceComputeRegionInstanceGroupManager() *schema.Resource {
 					},
 				},
 			},
+
+			"distribution_policy": &schema.Schema{
```

Comment:
The thing that I'm not totally sold on for this approach is that with a name like "distribution_policy", it should really be a list of policies rather than a list of zones.

As much as the nesting is weird for just one field, it does make the resource more maintainable in the future (especially once we start [generating](https://github.com/terraform-providers/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#autogenerated-resources) resources). If the API maintainers decide to add another field to that block for example, we'd then have to deprecate this field and make it into a nested block, which is a bit of a pain.

Happy to hear a rebuttal though :)

Reply:
I'm not against such a change.

Like I mentioned in the opening comments, I made the choices I did to present a good UX to the end-user: ultimately, this attribute is simply for saying which zones in a region should be used.

I guess I'd be curious about how easily users could splat out a list of zones into such a list of nested blocks.  I'm not sure I've seen that or done it myself.

Reply:
Alright, so I ended up putting you suggestion into action.  Now it's a list of objects, matching the structure of the API.

Reply:
Sorry to go back and forth on this- your comment about being able to splat is a good one. I asked around a bit and the suggestion I got that I like is to just have a top-level list of strings called `distribution_policy_zones`. That leaves room to deprecate it later in favor of a straight `distribution_policy` object in case the API adds more things, but makes a good UX for the time being.

Reply:
You're killin' me, smalls! :)

Alright, I'll go with that approach.